### PR TITLE
[#166240820] Validate signup

### PR DESCRIPTION
**Pull request by murediane**
#### What does this PR do?
Validate user signupInput
#### Description of Task to be completed?
* write unit tests
* Validate signup input  from the user

#### How should this be manually tested?
send data to be validated  request on auth/signup 
#### What are the relevant pivotal tracker stories?
[166240820](https://www.pivotaltracker.com/story/show/166240820)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,52 +1,53 @@
...
+		"morgan": "^1.9.1",
+		"pg": "^7.11.0",
+		"sequelize": "^5.8.7",
+		"swagger-node-express": "^2.1.3",
+		"swagger-ui-express": "^4.0.6"
+	},
+	"nyc": {
+		"per-file": true,
+		"check-coverage": true,
+		"branches": 50,
```
Comment (premlogic) [inline package.json]: The threshold should be 80 for branches and 100 for functions. May you change these parameters and make sure your tests are still passing?

File: src/middlewares/ValidateUser.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+/* eslint-disable no-tabs */
+/* eslint-disable no-unused-vars */
+import models from '../api/models/index';
+import signupValidator from '../helpers/validators/signupValidator';
+
+const { User } = models;
+/**
```
Comment (premlogic) [inline src/middlewares/ValidateUser.js:6]: Add a description of your classes in your ```JSdocs``` please

**Review by nkpremices — CHANGES_REQUESTED** [review]

**Review by alainburindi — COMMENTED** [review]
where is the relevant pivotal tracker story??

File: tests/middlewares/signupValidator.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  it('should return an error if the email is invalid', (done) => {
+    const { email, ...rest } = fakeUser;
+    chai
+      .request(app)
+      .post('/api/auth/signup')
+      .send(rest)
+      .end((err, res) => {
+        const { status, body } = res;
+        expect(status).to.equal(400);
+        expect(body).to.have.property('message');
```
Comment (alainburindi) [inline tests/middlewares/signupValidator.test.js]: what is the expected value?? specially for message

Comment (alainburindi) [reply tests/middlewares/signupValidator.test.js]: this concerns also the other tests

File: tests/middlewares/signupValidator.test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        done();
+      });
+  });
```
Comment (alainburindi) [inline tests/middlewares/signupValidator.test.js:58]: I don't see the test for `confirmPassword`

File: src/helpers/validators/signupValidator.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+import Joi from 'joi';
```
Comment (premlogic) [inline src/helpers/validators/signupValidator.js]: I have a file where I implement a logic like this one also. I think we can mix it with this one for us to have an easily understandable codebase

Comment (premlogic) [reply src/helpers/validators/signupValidator.js]: ```helpers/validation.schemas.js```

Comment (murediane) [reply src/helpers/validators/signupValidator.js]: I don't think it is an issue now since your PR is not merged yet, we will work on that while rebasing.

**Review by nkpremices — APPROVED** [review]

File: src/helpers/validators/signupValidator.js (hunk 5 above)
Comment (premlogic) [reply src/helpers/validators/signupValidator.js]: cool

File: tests/middlewares/signupValidator.test.js (hunk 3 above)
Comment (alainburindi) [reply tests/middlewares/signupValidator.test.js]: you have not yet worked on my feedbacks! yet they are pending for 5 days ago

**Review by alainburindi — APPROVED** [review]

File: tests/middlewares/signupValidator.test.js (hunk 3 above)
Comment (MemunaHaruna) [reply tests/middlewares/signupValidator.test.js]: @murediane please address @alainburindi's feedback.

File: .eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
+{
```
Comment (MemunaHaruna) [inline .eslintrc.json]: @murediane please, rebase with develop. `.eslintrc.json` is no longer on develop. Also, I don't understand why eslint config file is part of your PR. Please, ensure only files related to your PR show up in your commits.

File: package.json [hunk 7]
```diff
@@ -1,70 +1,71 @@
 {
```
Comment (MemunaHaruna) [inline package.json:1]: @murediane same with this file. I can't see any changes specific to your PR here.

File: src/middlewares/ValidateUser.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-tabs */
+/* eslint-disable no-unused-vars */
```
Comment (MemunaHaruna) [inline src/middlewares/ValidateUser.js]: @murediane why are we disabling eslint rules?

File: tests/middlewares/signupValidator.test.js (hunk 4 above)
Comment (MemunaHaruna) [reply tests/middlewares/signupValidator.test.js:58]: @murediane there's no test for `confirmPassword`. There's also no test for `existing email, existing username etc`

@MemunaHaruna Alain covered those test in signup test that is why I didn't repeat them.

**Review by MemunaHaruna — CHANGES_REQUESTED** [review]

File: src/middlewares/ValidateUser.js (hunk 8 above)
Comment (murediane) [reply src/middlewares/ValidateUser.js]: it is fixed  it was from before we set up rules

File: .eslintrc.json (hunk 6 above)
Comment (murediane) [reply .eslintrc.json]: done

File: package.json (hunk 7 above)
Comment (murediane) [reply package.json:1]: it's because of the package I had to install to validate data that was not in the develop package.jason

**Review by MemunaHaruna — APPROVED** [review]

**Pull request merged**